### ISSUE TITLE
kube-router v1.5.0 - most of requests are failing with DSR

### ISSUE THREAD

**Issue by opipenbe** [cloudnativelabs/kube-router#1328]
**What happened?**
After upgrading to a kube-router v1.5.0 most of requests are failing with DSR.

```
for i in {1..3}; do echo "Attempt: $i"; curl --connect-timeout 1 -I x.x.x.x; done
Attempt: 1
curl: (28) Connection timed out after 1000 milliseconds
Attempt: 2
HTTP/1.1 200 OK
Server: nginx/1.23.0
Date: Mon, 18 Jul 2022 13:07:10 GMT
Content-Type: text/html
Content-Length: 615
Last-Modified: Tue, 21 Jun 2022 14:25:37 GMT
Connection: keep-alive
ETag: "62b1d4e1-267"
Accept-Ranges: bytes

Attempt: 3
curl: (28) Connection timed out after 1001 milliseconds
```

There are no problems after building custom image without "Refactor ExternalIP Service Handling #1246" .

**What did you expect to happen?**
Requests to a service (DSR option: kube-router.io/service.dsr=tunnel) with external IP are successful.

**How can we reproduce the behavior you experienced?**
1. Configure DSR: https://github.com/cloudnativelabs/kube-router/blob/master/docs/dsr.md
2. Deploy application, service with an external IP and kube-router.io/service.dsr=tunnel:
```
kubectl create namespace dsr-tst
kubectl create deployment nginx --namespace dsr-tst --image=nginx
kubectl expose deployment nginx --namespace dsr-tst --type ClusterIP --port 80 --target-port=80
kubectl patch svc nginx --namespace dsr-tst -p '{"spec":{"externalIPs":["$EXTERNAL_IP"]}}'
kubectl annotate service nginx --namespace dsr-tst "kube-router.io/service.dsr=tunnel"
```

4. Test outside cluster:
`for i in {1..3}; do echo "Attempt: $i"; curl --connect-timeout 1 -I x.x.x.x; done`

**System Information**
 - Kube-Router Version (`kube-router --version`): v1.5.0
 - Kube-Router Parameters: --run-router, --run-service-proxy, --run-firewall=true, --bgp-graceful-restart=true, --advertise-external-ip=true, --hairpin-mode=true, --service-external-ip-range
 - Kubernetes Version (`kubectl version`) : tested with v1.23.7, v1.24.2
 - Cloud Type: on premise
 - Kubernetes Deployment Type: Kubeadm
 - Kube-Router Deployment Type:  DaemonSet
 - Cluster Size: 9 nodes
 - Tested with CRI-O and Docker runtime

**Comment by aauren** [comment]
@opipenbe - Thanks for the detailed report! I'll take a look at this early next week.

**Comment by aauren** [comment]
It took me a little while to replicate that issue in my setup. For some reason I wasn't able to effectively force the issue just by adding the DSR annotation. However, I was able to effectively replicate the issue by scaling up and/or down the deployment afterwards.

This makes sense to me, because essentially what was happening is that the services were failing a protocol check before they entered the service cleanup logic. This happened despite the fact that FW Marked services don't have a protocol. The prior implementation defaulted to UDP when the protocol wasn't recognized, while this wasn't really correct, it did at least allow the service cleanup to function.

Anyway, this should be fixed now with #1338 and I'll be cutting a 1.5.1 release soon that contains this and a few other fixes.

One note here, in my testing I found that sometimes there were artifacts left behind by the DSR setup of old services that were not cleaned. These seemed to persist even after this fix is applied. If you have services that have been affected by this problem, I would recommend doing a rolling restart of your nodes to ensure that all previous artifacts that might be left over are cleaned up.
